core/tools/messagehandler/backtracemodel.cpp

```diff
@@ -0,0 +1,131 @@
+/*
+  backtracemodel.cpp
+
+  This file is part of GammaRay, the Qt application inspection and
+  manipulation tool.
+
+  Copyright (C) 2010-2015 Klarälvdalens Datakonsult AB, a KDAB Group company, info@kdab.com
+  Author: Gábor Angyal <angyalgabor@outlook.com>
+
+  This program is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 2 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "backtracemodel.h"
+
+#include <tuple>
+
+using namespace std;
+using namespace GammaRay;
+
+BacktraceModel::BacktraceModel(QObject *parent)
+{
+}
+
+int BacktraceModel::rowCount(const QModelIndex &parent) const
+{
+  Q_UNUSED(parent);
```

Comment:
try to hook a ModelTester up to this model and it should spot some issues. Here for example, you should return 0 when parent is valid to indicate that this is not a tree model.